chainer/functions/normalization/decorrelated_batch_normalization.py

```diff
@@ -0,0 +1,305 @@
+                                  ' decorrelated batch normalization.')
+
+
+class FixedDecorrelatedBatchNormalizationFunction(function_node.FunctionNode):
+
+    def __init__(self, groups):
+        self.groups = groups
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 3)
+        x_type, mean_type, var_type = in_types
+        type_check.expect(
+            x_type.dtype.kind == 'f',
+            mean_type.dtype == x_type.dtype,
+            var_type.dtype == x_type.dtype,
+        )
+        type_check.expect(
+            x_type.ndim >= 2,
+        )
+
+    def forward(self, inputs):
+        self.retain_inputs((0, 1, 2))
+        x, mean, projection = inputs
+        x_shape = x.shape
+        b, c = x_shape[:2]
+        g = self.groups
+        C = c // g
+        spatial_axis, m = _calc_axis_and_m(x_shape, b, g)
+
+        if g > 1:
+            x = x.reshape((b * g, C) + x.shape[2:])
+        x_hat = x.transpose((1, 0) + spatial_axis).reshape(C, -1)
+
+        y_hat = projection.dot(x_hat - mean[:, None])
+
+        y = y_hat.reshape((C, b * g) + x.shape[2:]).transpose(
+            (1, 0) + spatial_axis)
+        if g > 1:
+            y = y.reshape((-1, c) + x.shape[2:])
+
+        return y,
+
+    def backward(self, indexes, grad_outputs):
+        x, mean, projection = self.get_retained_inputs()
+        gy,  = grad_outputs
+        f = FixedDecorrelatedBatchNormalizationGrad(self.groups)
+        return f.apply((x, mean, projection, gy))
+
+
+class FixedDecorrelatedBatchNormalizationGrad(function_node.FunctionNode):
+
+    def __init__(self, groups):
+        self.groups = groups
+
+    def forward(self, inputs):
+        self.retain_inputs(())
+        x, mean, projection, gy = inputs
+        gy_shape = gy.shape
+        b, c = gy_shape[:2]
+        g = self.groups
+        C = c // g
+        spatial_axis, m = _calc_axis_and_m(gy_shape, b, g)
+
+        if g > 1:
+            gy = gy.reshape((b * g, C) + gy.shape[2:])
+            x = x.reshape((b * g, C) + x.shape[2:])
+        x_hat = x.transpose((1, 0) + spatial_axis).reshape(C, -1)
+        gy_hat = gy.transpose((1, 0) + spatial_axis).reshape(C, -1)
+        gy_hat_pca = projection.T.dot(gy_hat)
+        gx = gy_hat_pca.reshape(
+            (C, b * g) + gy.shape[2:]).transpose((1, 0) + spatial_axis)
+        if g > 1:
+            gx = gx.reshape((-1, c) + gy.shape[2:])
+        rhs = x_hat - mean[Ellipsis, None]
+        gprojection = (x_hat - rhs).T.dot(gy_hat)
+        gmean = -gx[:, 0]
+        self.retain_outputs(())
+        return gx, gmean, gprojection
+
+    def backward(self, inputs, grad_outputs):
+        # TODO(crcrpar): Implement this.
+        raise NotImplementedError('Double backward is not implemented for'
+                                  ' fixed decorrelated batch normalizatin.')
+
+
+def decorrelated_batch_normalization(x, **kwargs):
+    """Decorrelated batch normalization function.
```

Comment:
Could you document the default values, similar to e.g. https://github.com/chainer/chainer/blob/v5.3.0/chainer/functions/normalization/batch_normalization.py#L597 ?

Reply:
Or actually, do you need `argument.parse_kwarg` at all?